idem_provider_azurerm/states/azurerm/resource/management_lock.py

```diff
@@ -729,3 +757,202 @@
 
     ret['comment'] = 'Failed to delete management lock {0}!'.format(name)
     return ret
+
+
+async def present_at_subscription_level(hub, ctx, name, lock_level, notes=None, owners=None, connection_auth=None,
```

Comment:
can we roll the resource and subscription level states into a generic "present" state? maybe handle a parameter that governs where the lock is applied.